nmap_dns.cc

```diff
@@ -855,7 +870,6 @@ static void connect_dns_servers() {
     serverI->write_busy = 0;
 
     nsock_connect_udp(dnspool, serverI->nsd, connect_evt_handler, NULL, (struct sockaddr *) &serverI->addr, serverI->addr_len, 53);
-    nsock_read(dnspool, serverI->nsd, read_evt_handler, -1, NULL);
```

Comment:
I have removed `nsock_read` from here and put it [there](https://github.com/nmap/nmap/pull/434/files#diff-1d4385f6f739c6e2c7e15e2e16a56d16R503) so that `request` can also be sent as the user data to `read_evt_handler`.